### ISSUE TITLE
Query invalid node type at position

### ISSUE THREAD

**Issue by G0V1NDS** [rest-nvim/tree-sitter-http#11]
With recent tree-sitter upgrade facing some error.

Information provided by healthcheck:
```
The following errors have been detected: ~
- ERROR http(highlights): ...-9033d5e/share/nvim/runtime/lua/vim/treesitter/query.lua:259: query: invalid node type at position 4 for language http
  http(highlights) is concatenated from the following files:
  | [ERROR]:"/Users/govind.r/.local/share/nvim/lazy/nvim-treesitter/queries/json/highlights.scm", failed to load: ...-9033d5e/share/nvim/runtime/lua/vim/treesitter/query.lua:259: query: invalid node type at position 4 for language http
```

<img width="1440" alt="Screenshot 2023-04-21 at 7 56 10 AM" src="https://user-images.githubusercontent.com/14184045/233526294-e6b9b2f6-6189-487f-8b17-06b67c3ee872.png">

**Comment by illia-danko** [comment]
The same for me. To reproduce the issue just create an empty http file and open it. Reproduced on
```
NVIM v0.9.0
Build type: Release
LuaJIT 2.1.0-beta3
```

**Comment by jackpinto** [comment]
Same Here

```
NVIM v0.9.0
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

**Comment by gustavoe** [comment]
Same issue. Any workarounds?

```
NVIM v0.9.0
Build type: Release
LuaJIT 2.1.0-beta3
```

**Comment by G0V1NDS** [comment]
@gustavoe Workaround is present in this linked issue https://github.com/rest-nvim/rest.nvim/issues/191
You can unblock yourself by using code prior to this issue, move to below mentioned commit for `'nvim-treesitter/nvim-treesitter'`
```
commit hash: eedc5198a1b4bb1b08ae6d4f64f3d76e376957aa
```

**Comment by NTBBloodbath** [comment]
Hey, I'm going to try to fix the `highlights.scm` file later today. Sorry it took too long, I've been a bit busy nowadays!

Cheers

**Comment by NTBBloodbath** [comment]
Debugging a `highlights.scm` file is really annoying as error messages are far from being helpful so this is going to take longer than expected.

**Comment by illia-danko** [comment]
@NTBBloodbath no worries, take you time. For now, i've reverted tree-sitter to eedc5198a1b4bb1b08ae6d4f64f3d76e376957aa as a workaround.

**Comment by NTBBloodbath** [comment]
Finally good news, queries seems to be working again 🎉 
![image](https://user-images.githubusercontent.com/36456999/236344461-4a04fb3d-b0f2-4d47-9e01-eb65a4a05049.png)

I'm going to close this as soon as nvim-treesitter gets the update and the issue gets fixed. Sorry for the delay, had a busy weekend and I didn't have enough time to sit and debug this thing haha

**Comment by yuukilla** [comment]
i dont know if im supposed to open new issue or i can comment down here
but highlight does not seem to work correctly 
![image](https://user-images.githubusercontent.com/56758452/236350737-df669891-f7f5-4a22-9372-2c7c51af9632.png)

**Comment by NTBBloodbath** [comment]
> i dont know if im supposed to open new issue or i can comment down here but highlight does not seem to work correctly ![image](https://user-images.githubusercontent.com/56758452/236350737-df669891-f7f5-4a22-9372-2c7c51af9632.png)

Highlighting is something that has to do with your Neovim colorscheme in most cases like header names, please refer to the `highlights.scm` file in the `queries` directory to know what the highlighting groups are (e.g. `@foo`).

Also make sure to have JSON tree-sitter parser installed for JSON bodies highlighting as it is injected into HTTP one.

Cheers

**Comment by NTBBloodbath** [comment]
> i dont know if im supposed to open new issue or i can comment down here but highlight does not seem to work correctly ![image](https://user-images.githubusercontent.com/56758452/236350737-df669891-f7f5-4a22-9372-2c7c51af9632.png)

Highlighting is something that has to do with your Neovim colorscheme in most cases like header names, please refer to the `highlights.scm` file in the `queries` directory to know what the highlighting groups are (e.g. `@foo`).

Also make sure to have JSON tree-sitter parser installed for JSON bodies highlighting as it is injected into HTTP one.

Cheers

**Comment by joshuahicks** [comment]
Thanks for the fix @NTBBloodbath, works great 🙏

**Comment by jackpinto** [comment]
Works like a charm, thanks, great work @NTBBloodbath !

**Comment by NTBBloodbath** [comment]
Now that nvim-treesitter got the upstream changes I'll be closing this along with the related issue in the rest.nvim plugin repository.

Thanks for waiting and sorry for the delay!
